Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -78,26 +86,49 @@ ResourcePack::ResourcePack(const std::string& path) : m_path(path)
 
   unzGoToFirstFile(file);
 
+  std::array<ResourceType, 2> resource_types = {ResourceType::TEXTURE, ResourceType::DYNAMIC_INPUT};
+
   do
   {
     std::string filename(256, '\0');
 
-    unz_file_info texture_info;
-    unzGetCurrentFileInfo(file, &texture_info, filename.data(), static_cast<u16>(filename.size()),
+    unz_file_info file_info;
+    unzGetCurrentFileInfo(file, &file_info, filename.data(), static_cast<u16>(filename.size()),
                           nullptr, 0, nullptr, 0);
 
-    if (filename.compare(0, 9, "textures/") != 0 || texture_info.uncompressed_size == 0)
-      continue;
-
-    // If a texture is compressed and the manifest doesn't state that, abort.
-    if (!m_manifest->IsCompressed() && texture_info.compression_method != 0)
+    for (auto resource_type : resource_types)
     {
-      m_valid = false;
-      m_error = "Texture " + filename + " is compressed!";
-      return;
-    }
+      std::string zip_directory_root;
+      std::string load_directory_root;
+      if (resource_type == ResourceType::TEXTURE)
+      {
+        zip_directory_root = "textures/";
+        load_directory_root = TEXTURE_PATH;
+      }
+      else
+      {
+        zip_directory_root = "dynamic_input/";
+        load_directory_root = DYNAMIC_INPUT_PATH;
+      }
+
+      if (filename.compare(0, zip_directory_root.size(), zip_directory_root) != 0 ||
+          file_info.uncompressed_size == 0)
+        continue;
```

Comment:
Since this `if` condition spans multiple lines, curlies should be used for the body.